scripts/install-jstz-cli.sh

```diff
@@ -72,22 +70,30 @@ EOF
 
     echo "Configuring \`jstz\` alias..."
 
-    shell=$(basename "$SHELL")
-    case "$shell" in
-        "bash")
-            shellrc="$HOME/.bashrc"
-            ;;
-        "zsh")
-            shellrc="$HOME/.zshrc"
-            ;;
-        *)
-            cat 1>&2 << EOF
-Unsupported shell: $shell. 
-Please manually add the alias to your shell's configuration file.
-    $shell_alias
-EOF
-            exit 1
-    esac
+    current_shell=$(ps -p $$ -ocomm=)
+
+    echo "SHELL: $SHELL"
+    echo "Detected shell: $current_shell"
+
+    if [[ $SHELL == */bash ]]; then
+        shellrc="$HOME/.bashrc"
+    elif [[ $SHELL == */zsh ]]; then
+        # Respect the ZDOTDIR variable if set, defaulting to $HOME if not
+        shellrc="${ZDOTDIR:-$HOME}/.zshrc"
+    else
+        echo >&2 "Unsupported shell: $SHELL. Please manually add the following alias to your shell's configuration file."
+        echo >&2 "    $shell_alias"
+        exit 1
+    fi
+
+    if ! [ -w "$shellrc" ]; then
+        echo "Warning: $shellrc is not writable. Please manually add the following alias to your shell's configuration file:"
+        echo "$shell_alias"
+
+        echo "Once you have added the alias, run the following command to reload the configuration file in your default shell:"
+        echo "    source $shellrc"
+        exit 1  # Exit the script with an error status
+    fi
```

Comment:
Rebase artifact?

Reply:
yeah, should have been part of the previous PR